src/Servers/HttpSys/src/AsyncAcceptContext.cs

```diff
@@ -69,15 +71,16 @@ private static void IOCompleted(AsyncAcceptContext asyncContext, uint errorCode,
                     // points to it we need to hook up our authentication handling code here.
                     try
                     {
-                        var nativeContext = asyncContext._nativeRequestContext;
+                        var requestContext = asyncContext._requestContext;
 
-                        if (server.ValidateRequest(nativeContext) && server.ValidateAuth(nativeContext))
+                        if (server.ValidateRequest(requestContext) && server.ValidateAuth(requestContext))
                         {
                             // It's important that we clear the native request context before we set the result
                             // we want to reuse this object for future accepts.
-                            asyncContext._nativeRequestContext = null;
+                            asyncContext._requestContext = null;
+
+                            requestContext.Initialize();
```

Comment:
Why is Initialize being called in two different places? Once in Accept and once in Execute?

Reply:
I'm lazy and didn't want to update the tests

Reply:
Shouldn't you be able to remove the call from Execute? It's always called after Accept.

Reply:
I really wanted to just do it in execute and not accept but some unit tests were failing because they didn't test the entire pipeline. I don't mind removing it from one place. Personally I think we should do it all on the worker thread.

Reply:
Fixed